api/app/weather_models/env_canada.py

```diff
@@ -470,6 +492,24 @@ def _process_prediction(self,  # pylint: disable=too-many-arguments
         # Add this prediction to the session (we'll commit it later.)
         self.session.add(station_prediction)
 
+    def _calculate_delta_precip(self, station, model_run, prediction, station_prediction):
+        """ Calculate the station_prediction's delta_precip based on the previous precip
+        prediction for the station
+        """
+        results = self.session.query(WeatherStationModelPrediction).\
+            filter(WeatherStationModelPrediction.station_code == station.code).\
+            filter(WeatherStationModelPrediction.prediction_model_run_timestamp_id == model_run.id).\
+            filter(WeatherStationModelPrediction.prediction_timestamp < prediction.prediction_timestamp).\
+            order_by(WeatherStationModelPrediction.prediction_timestamp.desc()).\
+            limit(1).first()
```

Comment:
I think `first()` can go away in this case right?

Reply:
No ```first()``` still has a use here - you still want to get the 1st record in the query. Otherwise it's returning a query collection for you.